mail/api.py

```diff
@@ -130,21 +147,38 @@ def send_batch(cls, subject, body, recipients,  # pylint: disable=too-many-argum
             params['recipient-variables'] = json.dumps({email: {} for email in chunk})
             if sender_address:
                 params['from'] = sender_address
-            responses.append(
-                cls._mailgun_request(
+
+            if settings.MAILGUN_RECIPIENT_OVERRIDE is not None:
+                original_recipients_chunk = original_recipients
+            else:
+                original_recipients_chunk = chunk
+
+            try:
+                response = cls._mailgun_request(
                     requests.post,
                     'messages',
                     params,
-                    sender_name=sender_name
+                    sender_name=sender_name,
+                )
+                if response.status_code == status.HTTP_401_UNAUTHORIZED:
+                    message = "Mailgun API keys not properly configured."
+                    log.error(message)
+                    raise ImproperlyConfigured(message)
```

Comment:
I don't think this works the way you expect. If you raise this exception here, it will be caught by the `except Exception as exception:` line a few lines later. Then the code will stuff this `ImproperlyConfigured` exception into the list of responses, and merrily continue iterating over chunks. It would be better to have the `ImproperlyConfigured` exception actually raise, instead of being caught, so that this code doesn't try to contact Mailgun more than once. If the API keys aren't configured, contacting Mailgun will never succeed!